File: frontend/src/model/feedback.ts
/** Models for table: feedback (ids are number / BIGINT) */

export class FeedbackRes {
  id: number = 0;
  orgId: number = 0;
  locationId: number = 0;
  customer: string = "";
  invoice: string = "";
  staff: string = "";
  outlet: string = "";
  date: string = "";
  rating: number = 0;
  nps: number = 0;
  channel: string = "";
  status: string = "";
  comment: string = "";
  createdby: string = "";
  createdon: string = "";
  updatedby: string = "";
  updatedon: string = "";
}

export class CreateFeedbackReq {
  orgId: number = 0;
  locationId: number = 0;
  customer: string = "";
  invoice: string = "";
  staff: string = "";
  outlet: string = "";
  date: string = "";
  rating: number = 0;
  nps: number = 0;
  channel: string = "";
  status: string = "";
  comment: string = "";
}

export class CreateFeedbackRes extends FeedbackRes {}

export class UpdateFeedbackReq {
  id: number = 0;
  orgId: number = 0;
  locationId: number = 0;
  customer: string = "";
  invoice: string = "";
  staff: string = "";
  outlet: string = "";
  date: string = "";
  rating: number = 0;
  nps: number = 0;
  channel: string = "";
  status: string = "";
  comment: string = "";
}

export class UpdateFeedbackRes extends FeedbackRes {}

export class GetFeedbackReq {
  id: number = 0;
  orgId: number = 0;
}

export class GetFeedbackRes extends FeedbackRes {
  errorMessage: string = "";
}

export class ListFeedbackReq {
  orgId: number = 0;
  locationId: number = 0;
  search: string = "";
  page: number = 1;
  pageSize: number = 50;
}

export class ListFeedbackRes {
  items: FeedbackRes[] = [];
  total: number = 0;
  errorMessage: string = "";
}

export class DeleteFeedbackReq {
  id: number = 0;
  orgId: number = 0;
}

export class DeleteFeedbackRes {
  id: number = 0;
  success: boolean = false;
  errorMessage: string = "";
}

File: backend/Krios/krios/Models/Feedback.cs
namespace Krios.Models.Krios
{
    public class Feedback
    {
        public long id { get; set; }
        public long orgId { get; set; }
        public long locationId { get; set; }
        public string customer { get; set; } = "";
        public string invoice { get; set; } = "";
        public string staff { get; set; } = "";
        public string outlet { get; set; } = "";
        public DateTime? date { get; set; }
        public long rating { get; set; }
        public long nps { get; set; }
        public string channel { get; set; } = "";
        public string status { get; set; } = "";
        public string comment { get; set; } = "";
        public string createdby { get; set; } = "";
        public DateTime? createdon { get; set; }
        public string updatedby { get; set; } = "";
        public DateTime? updatedon { get; set; }
    }

    public class FeedbackSelectReq
    {
        public long id { get; set; }
        public long orgId { get; set; }
        public long locationId { get; set; }
        public string search { get; set; } = "";
        public string status { get; set; } = "";
    }

    public class FeedbackDeleteReq
    {
        public long id { get; set; }
        public long orgId { get; set; }
    }
}